sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1235,6 +1238,115 @@ static ipv6_addr_t *_src_addr_selection(gnrc_netif_t *netif,
         return &netif->ipv6.addrs[idx];
     }
 }
+
+#if IS_USED(MODULE_GNRC_SIXLOWPAN_CTX)
+static bool _ctx_match(const gnrc_sixlowpan_ctx_t *ctx,
+                       const ipv6_addr_t *prefix, uint8_t prefix_len)
+{
+    return (ctx != NULL) &&
+           (ctx->prefix_len == prefix_len) &&
+           (ipv6_addr_match_prefix(&ctx->prefix, prefix) >= prefix_len);
+}
+
+/* configure 6LoWPAN compression context */
+static void _update_6ctx(const ipv6_addr_t *prefix, uint8_t prefix_len,
+                         uint32_t valid)
+{
+    gnrc_sixlowpan_ctx_t *ctx = gnrc_sixlowpan_ctx_lookup_addr(prefix);
+    uint8_t cid = 0;
+
+    if (!_ctx_match(ctx, prefix, prefix_len)) {
+        /* While the context is a prefix match, the defined prefix within the
+         * context does not match => use new context */
+        ctx = NULL;
+    }
+    else {
+        cid = ctx->flags_id & GNRC_SIXLOWPAN_CTX_FLAGS_CID_MASK;
+    }
+    /* find first free context ID */
+    if (ctx == NULL) {
+        while (((ctx = gnrc_sixlowpan_ctx_lookup_id(cid)) != NULL) &&
+               !_ctx_match(ctx, prefix, prefix_len)) {
+            cid++;
+        }
+    }
+    if (cid < GNRC_SIXLOWPAN_CTX_SIZE) {
+        DEBUG("GNRC util: add compression context %u for prefix %s/%u\n", cid,
+              ipv6_addr_to_str(addr_str, prefix, sizeof(addr_str)),
+              prefix_len);
+        gnrc_sixlowpan_ctx_update(cid, (ipv6_addr_t *)prefix, prefix_len,
+                                  valid / (60 * MS_PER_SEC),
+                                  true);
+    }
+}
+#else
+static void _update_6ctx(const ipv6_addr_t *prefix, uint8_t prefix_len,
+                         uint32_t valid)
+{
+    (void)prefix;
+    (void)prefix_len;
+    (void)valid;
+}
+#endif
```

Comment:
Last thought (and I am aware that this is more or less asking you to clean up my mess :sweat_smile:): Would it make sense to move these as convenience functions to `net/gnrc/sixlowpan/ctx.h`?

Reply:
Are those used anywhere else?

Reply:
Not really, but they are more about manipulating the context store than changing a `gnrc_netif` (they don't even use the `gnrc_netif_t` type anywhere) + you surround them with a module-specific `#ifdef` anywhere, so it seems cleaner to put them in the module they belong to ;-).

Reply:
You can make them `static inline`, so using them should not change the binary.

Reply:
```C
        DEBUG("GNRC util: add compression context %u for prefix %s/%u\n", cid,
              ipv6_addr_to_str(addr_str, prefix, sizeof(addr_str)),
              prefix_len);
```

won't work as `static inline` if there is no global `addr_str` buffer

Reply:
> ```c
> ```c
>         DEBUG("GNRC util: add compression context %u for prefix %s/%u\n", cid,
>               ipv6_addr_to_str(addr_str, prefix, sizeof(addr_str)),
>               prefix_len);
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> won't work as static inline if there is no global addr_str buffer
> ```

Do we really need it?

Reply:
If you think it is valuable, we can also add a `DEBUG` statement [here](https://github.com/RIOT-OS/RIOT/pull/16672/files#diff-2ad823469e86d3772efecaa0f4bb5afcc1d6d039af9cc28db3535fcd292e28a6R1342)

```c
        DEBUG("GNRC util: attempting to add compression context %u for prefix %s/%u\n", cid,
              ipv6_addr_to_str(addr_str, prefix, sizeof(addr_str)),
              prefix_len);
```

Reply:
Ok, moved it there and added some documentation.